yoti_python_sdk/profile.py

```diff
@@ -19,6 +20,12 @@ def __init__(self, profile_attributes):
                         field.content_type
                     )
 
+                    # first condition will be removed in v3.0.0, so selfie also returns an Image object
+                    if field.name != config.ATTRIBUTE_SELFIE and \
+                            (field.content_type == Protobuf.CT_JPEG \
+                             or field.content_type == Protobuf.CT_PNG):
+                        value = Image(field.value, field.content_type)

```

Comment:
It probably doesn't make much difference now, but could the conversion to Image object happen in `Protobuf().value_based_on_content_type`, then have this use the image object to get the image content, e.g.

```python
if field.name == config.ATTRIBUTE_SELFIE and isinstance(value, Image):
    value = value.content
```

Then we just remove this check completely in the major release? _(I think it's fine as is, but thought I'd mention for consideration)_

You might need to move Image object conversion into `Protobuf().value_based_on_content_type` for document images

Reply:
That would mean I'd have to do it in two separate places - as we have one for the old profile, and one for the new one. But I still think it's a good idea to ensure we don't miss this in the major. Will implement